src/plugins/intel_cpu/CMakeLists.txt

```diff
@@ -9,7 +9,7 @@ endif()
 set(TARGET_NAME "openvino_intel_cpu_plugin")
 
 if(CMAKE_COMPILER_IS_GNUCC)
-    ie_add_compiler_flags(-Wno-all)
+    ie_add_compiler_flags(-Wno-all -Werror -Wundef -Wreturn-type -Wunused-variable -Wunused-value -Wswitch -Wformat -Wformat-security)
```

Comment:
why not to just fix `-Wall` ?

Reply:
> why not to just fix `-Wall` ?

With `-Wall` the project will not compile. There will be errors related to the `-Werror=sign-compare` and `-Werror=reorder` flags that are not set in other build configurations and CI checks.